service/src/main/java/bio/terra/cbas/controllers/RunsApiController.java

```diff
@@ -22,28 +24,65 @@ public class RunsApiController implements RunsApi {
 
   private final CromwellServerConfiguration cromwellConfig;
 
+  private RunState convertToRunState(String workflowStatus) {
+    return switch (workflowStatus) {
+      case "On Hold" -> RunState.PAUSED;
+      case "Submitted" -> RunState.QUEUED;
+      case "Running" -> RunState.RUNNING;
+      case "Aborting" -> RunState.CANCELING;
+      case "Aborted" -> RunState.CANCELED;
+      case "Succeeded" -> RunState.COMPLETE;
+      case "Failed" -> RunState.EXECUTOR_ERROR;
+      default -> RunState.UNKNOWN;
+    };
+  }
+
+  private Date convertToDate(OffsetDateTime submissionTimestamp) {
+    if (submissionTimestamp != null) {
+      return new Date(submissionTimestamp.toInstant().toEpochMilli());
+    }
+
+    return null;
+  }
+
+  private LogRunRequest convertToLogRunRequest(WorkflowQueryResult queryResult) {
```

Comment:
This feels more like a `LogRunResponse` than a request?

Reply:
Good point. I will rename it.